src/RamOperation.h

```diff
@@ -263,74 +267,126 @@ class RamParallelScan : public RamScan, public RamAbstractParallel {
         RamRelationOperation::print(os, tabpos + 1);
     }
 };
-
 /**
  * @class Relation Scan with Index
  * @brief An abstract class for performing indexed operations
  */
 class RamIndexOperation : public RamRelationOperation {
 public:
-    RamIndexOperation(std::unique_ptr<RamRelationReference> r, int ident,
-            std::vector<std::unique_ptr<RamExpression>> queryPattern, std::unique_ptr<RamOperation> nested,
-            std::string profileText = "")
+    RamIndexOperation(std::unique_ptr<RamRelationReference> r, int ident, RamPattern queryPattern,
+            std::unique_ptr<RamOperation> nested, std::string profileText = "")
             : RamRelationOperation(std::move(r), ident, std::move(nested), std::move(profileText)),
               queryPattern(std::move(queryPattern)) {
-        assert(getRangePattern().size() == getRelation().getArity());
-        for (const auto& pattern : queryPattern) {
+        assert(getRangePattern().first.size() == getRelation().getArity() &&
+                getRangePattern().second.size() == getRelation().getArity());
+        for (const auto& pattern : queryPattern.first) {
+            assert(pattern != nullptr && "pattern is a null-pointer");
+        }
+        for (const auto& pattern : queryPattern.second) {
             assert(pattern != nullptr && "pattern is a null-pointer");
         }
     }
 
     /**
      * @brief Get range pattern
-     * @return A std::vector of pointers to RamExpression objects
```

Comment:
Can you mention lower / upper bounds here? Otherwise, the message is too cryptic.